src/bitcoin/mod.rs

```diff
@@ -172,9 +171,28 @@ impl BitcoinInterface for d::BitcoinD {
                     if block_height > 6 {
                         spent.push((*op, *txid, tx.block_time.expect("Spend is confirmed")))
                     }
-                } else {
-                    // TODO: handle the case where new transaction which txid is not the
-                    // coin spending_txid, spent the coin.
+                } else if !tx.conflicting_txs.is_empty() {
+                    for txid in tx.conflicting_txs.clone() {
+                        let tx: Option<&d::GetTxRes> = match cache.get(&txid) {
+                            Some(tx) => tx.as_ref(),
+                            None => {
+                                let tx = self.get_transaction(&txid);
+                                cache.insert(txid, tx);
+                                cache.get(&txid).unwrap().as_ref()
+                            }
+                        };
+                        if let Some(tx) = tx {
+                            if let Some(block_height) = tx.block_height {
+                                if block_height > 6 {
```

Comment:
Same comment as in previous commit. I think we should remove the check for more than 6 confirmations.